### PR TITLE
adding helper filters url and static

### PR DESCRIPTION
**Pull request by samuelcolvin**
(I realised after starting this that it's slightly more controversial than I thought as `url` already existed as a function in the global environment context; still, I think this is clearer and a more canonical approach)

If others agree with this I'll add unit tests and update docs.

The point is to have two filters which can be used thus:
* `{{ 'get-user'|url(id=123) }}` (or just `{{ 'get-users'|url }}` if there are no variables in the url)
* `{{ 'styles.css'|static }}`

this uses jinja's context filter approach which is clearer than using closure and having a locally defined filter function.

I started this as they are methods in `aiohttp-devtools -> start` that I thought should be moved to a library. Wasn't even aware of the `url` function until now as it wasn't very clearly documented and wasn't implemented canonically.

-----------------------------

### Update

Now uses `contextfunction` for `url` and `static` so using a url would be just `{{ url('get-user', id=123) }}` and static `{{ static('styles.css') }}`.

### CODE REVIEW

**Comment by samuelcolvin** [comment]
bump. Anyone?

**Comment by jettify** [comment]
@asvetlov could you please take a look?

**Comment by asvetlov** [comment]
Sorry for pause.
I have no strong opinion.
We should document `url()` function, sure.
Regarding to `{{ 'get-user'|url(id=123) }}` syntax -- it looks weird for me.
@samuelcolvin what other libraries use such filters? You say this approach is canonical but honestly I don't sure if I want to use this format.

**Comment by samuelcolvin** [comment]
Ok, perhaps better to use a [context function](http://jinja.pocoo.org/docs/2.9/api/#jinja2.contextfunction) instead of context filter. 

That way urls would be at like `{{ url('get-user', id=123) }}` but the code would be more canonical.

I'll update the PR now.

**Comment by samuelcolvin** [comment]
Ok, I've switched static to be a function too to provide a consistent API. For me this is ready. 

@asvetlov let me know if there's anything you want tweaking?

**Comment by asvetlov** [comment]
Will review it soon, sure.
I have very tight schedule these days.

File: aiohttp_jinja2/helpers.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+useful context functions, see
+http://jinja.pocoo.org/docs/dev/api/#jinja2.contextfunction
+"""
+import jinja2
+
+
+@jinja2.contextfunction
+def reverse_url(context, __route_name, **parts):
+    """
+    filter for generating urls, see
```
Comment (asvetlov) [inline aiohttp_jinja2/helpers.py]: We don't use autodoc (and I reluctant to do it).

From my perspective sphinx markup should be placed in `.rst` files, docstrings should be relative small and don't contain any special symbols: they are invented for helping human reading, not for parsing by automated tool.
The source code reading is not comfortable If docstring is longer than function source itself.

Open CPython source code: it follows the same rule

**Comment by samuelcolvin** [comment]
updated as per @asvetlov's comments, sorry for delay.

**Pull request merged**